bftengine/src/bftengine/SimpleClientImp.cpp

```diff
@@ -108,6 +111,21 @@ class SimpleClientImp : public SimpleClient, public IReceiver {
   concordlogger::Logger logger_ = concordlogger::Log::getLogger("concord.bft.client");
 };
 
+bool SimpleClientImp::isSystemReady() const {
+  uint16_t connectedReplicasNum = 0;
+  if (communication_->isRunning()) {
+    for (uint16_t rid : replicas_) {
+      if (communication_->getCurrentConnectionStatus(rid) == ConnectionStatus::Connected) connectedReplicasNum++;
```

Comment:
Unfortunately, I don't think this works. The implementation of getCurrentConnectionStatus in TLS is [incorrect](https://github.com/vmware/concord-bft/blob/master/communication/src/TlsTCPCommunication.cpp#L1287). I discovered this when working on a refactoring a couple months ago. Unfortunately, I also did not have time to finish that (it's about 95%), so the fix is not in. 

You may be able to take my fix although you may need to modify it to fit the exact names in the current code base: 

```C++
ConnectionStatus TlsTCPCommunication::TlsTcpImpl::getCurrentConnectionStatus(const NodeNum id) const {
  std::lock_guard<std::mutex> lock(connectionsGuard_);
  if (connections_.count(id) == 1) {
    return ConnectionStatus::Connected;
  }
  return ConnectionStatus::Disconnected;
}
```

Reply:
Thanks for the observation, Andrew. I've also noticed that this functionality is not implemented, that's why prefixed this PR with 'WIP'.